config/main.py

```diff
@@ -1357,12 +1357,67 @@ def apply_patch(ctx, patch_file_path, format, dry_run, ignore_non_yang_tables, i
             patch_as_json = json.loads(text)
             patch = jsonpatch.JsonPatch(patch_as_json)
 
+        results = {}
         config_format = ConfigFormat[format.upper()]
-        GenericUpdater().apply_patch(patch, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+        if multi_asic.is_multi_asic():
+            # Initialize a dictionary to hold changes categorized by ASIC
+            changes_by_asic = {}
 
+            # Function to extract ASIC identifier from the change path
+            def extract_asic_id(path):
+                start = path.find("/") + 1
+                end = path.find("/", start)
+                return path[start:end], path[end:]  # Also return the modified path without ASIC ID
+
+            # Function to apply patch for a single ASIC.
+            def apply_patch_for_asic(asic_changes):
+                asic_id, changes = asic_changes
+
+                # Replace localhost to empty string which is db definition of Host
+                if asic_id.lower() == "localhost":
+                    asic_id = ""
+
+                try:
+                    # Call apply_patch with the ASIC-specific changes and predefined parameters
+                    GenericUpdater(namespace=asic_id).apply_patch(jsonpatch.JsonPatch(changes), config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+                    results[asic_id] = {"success": True, "message": "Success"}
+
+                    log.log_notice(f"'apply-patch' executed successfully for {asic_id} by {changes}")
+                except Exception as e:
```

Comment:
>except Exception [](http://example.com/codeflow?start=16&length=16)

There is another outer scope to 'except Exception'. Could you simplify? will one layer be enough?

Reply:
This is only for log purpose, but exception will be tracking in results, we will check any failure in results. This will not block other ASIC apply process.

Reply:
It is still arguable wether or not "This will not block other ASIC apply process."

Let's return an error code from function `apply_patch_for_scope`, and we can temporarily implement continuing other ASIC if one fails.

Reply:
If we just log and record the error into result dict, then it will swallow error instead of breaking the following ASIC apply.